File: lib_base/src/main/java/world/share/lib_base/data/impl/LocalDataSource.java
package world.share.lib_base.data.impl;


import world.share.baseutils.ToastUtil;
import world.share.baseutils.threadutil.LifeRunnable;
import world.share.baseutils.threadutil.ThreadX;
import world.share.lib_internet.local.BaseLocalData;

/**
 * @author mac
 */
public class LocalDataSource implements BaseLocalData {

    public void home() {
        ThreadX.getInstance().execute(new LifeRunnable() {
            @Override
            public void start() {
                super.start();
                ToastUtil.show("正在登录...");
            }

            @Override
            public void running() {
                try {
                    Thread.sleep(3000);
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }
            }

            @Override
            public void end() {
                super.end();
                ToastUtil.show("登录结束");
            }

        });
    }

}

File: lib_base/src/main/java/world/share/lib_base/data/LocalDataSource.kt
package world.share.lib_base.data

import world.share.lib_base.bean.UserBean

/**
 * @author wan
 * 创建日期：2021/10/15
 * 描述：本地数据提供接口
 */
interface LocalDataSource {

    fun getLocalData()

    fun saveUserData(userBean: UserBean)

}